spec/eof.md

```diff
@@ -161,10 +124,25 @@ Code executing within an EOF environment will behave differently than legacy cod
 - The instruction `JUMPDEST` is renamed to `NOP` and remains charging 1 gas without any effect.
     - Note: jumpdest-analysis is not performed anymore.
 - EOF contract may not deploy legacy code (it is naturally rejected on the code validation stage)
-- Legacy creation transactions (any tranactions with empty `to`) are invalid in case `data` contains EOF code (starts with `EF00` magic)
-- If instructions `CREATE` and `CREATE2` have EOF code as initcode (starting with `EF00` magic)
-    - deployment fails (returns 0 on the stack)
-    - caller's nonce is not updated and gas for initcode execution is not consumed
+
+#### Creation transactions
+
+Creation transactions (tranactions with empty `to`), with `data` containing EOF code (starting with `EF00` magic) are interpreted as having an EOF `initcontainer` in the `data` and:
+
+- intrinsic gas cost rules and limits defined in EIP-3860 for legacy creation transaction apply. The entire `initcontainer` is treated as initcode
+- **the initcontainer and all its subcontainers are validated recursively**
+- it is checked if the initcode container has its `len(data_section)` equal to `data_size`, i.e. data section content is exactly as the size declared in the header (see [Data section lifecycle](#data-section-lifecycle))
+- transaction fails if any of those checks were invalid (gas for initcode execution is not consumed. Only intrinsic creation transaction costs were consumed)
+- execute the container in "initcode-mode" and deduct gas for execution
+    - calculate `new_address` as `keccak256(sender || sender_nonce)[12:]`
```

Comment:
Can address be calculated with CREATE2-like scheme here instead? as we're anyway applying special EOF-rules to this kind of transaction.

Reply:
I guess we can salt into data section, next to calldata :see_no_evil:

Reply:
But then how do you fetch it and apply? Note that salt is an argument to TXCREATE, which is implicit here, we start execution just after the moment where it "would-have-been".

You either need to move it to RETURNCONTRACT or have a ugly special rule, that it is first bytes of the `data_section`, and the creation logic peeks there.

Reply:
Yeah seems even uglier than I thought, probably not worth it, let's stick with the nonce-based scheme.